app/src/main/java/org/p2p/wallet/auth/gateway/repository/GatewayServiceRemoteRepository.kt

```diff
@@ -52,6 +72,48 @@ class GatewayServiceRemoteRepository(
             otpConfirmationCode = otpConfirmationCode
         )
         val response = api.confirmRegisterWallet(request)
-        mapper.fromNetwork(response)
+        createWalletMapper.fromNetwork(response)
+    }
+
+    override suspend fun restoreWallet(
+        solanaPublicKey: Base58String,
+        solanaPrivateKey: Base58String,
+        phoneNumber: String,
+        channel: OtpMethod
+    ): GatewayServiceStandardResponse = withContext(dispatchers.io) {
+        val request = restoreWalletMapper.toRestoreWalletNetwork(
+            userPublicKey = solanaPublicKey,
+            userPrivateKey = solanaPrivateKey,
+            phoneNumber = phoneNumber,
+            channel = channel
+        )
+        val response = api.restoreWallet(request)
+        restoreWalletPublicKey = solanaPublicKey
+        createWalletMapper.fromNetwork(response)
+    }
+
+    override suspend fun confirmRestoreWallet(
+        solanaPublicKey: Base58String,
+        solanaPrivateKey: Base58String,
+        phoneNumber: String,
+        otpConfirmationCode: String
+    ): ConfirmRestoreWalletResponse = withContext(dispatchers.io) {
+        if (solanaPublicKey != restoreWalletPublicKey) {
+            throw RestoreWalletPublicKeyError(
+                expectedPublicKey = restoreWalletPublicKey?.base58Value,
+                actualPublicKey = solanaPublicKey.base58Value
+            )
+                .also { Timber.i(it) }
+        }
+
+        val request = restoreWalletMapper.toConfirmRestoreWalletNetwork(
+            userPublicKey = solanaPublicKey,
+            userPrivateKey = solanaPrivateKey,
+            phoneNumber = phoneNumber,
+            otpConfirmationCode = otpConfirmationCode
+        )
+        val response = api.confirmRestoreWallet(request)
+        createWalletMapper.fromNetwork(response)
+            .also { resetTemporaryPublicKeyTimer.stopTimer() }
```

Comment:
Is it the only place stop timer is called?
What if user didn't call `confirmRestoreWallet`? 
The timer will still work even if user will be on the different screen?

Reply:
he can't enter the app (home screen) unless he called confirmRestoreWallet, so timer would be stopped anyways
but yes, until he enters OTP sms code during logging in - the timer works